src/Admin/Pool.php

```diff
@@ -361,20 +369,25 @@ public function getAdminByClass($class)
             // NEXT_MAJOR : remove the previous `trigger_error()` call, the `return null` statement,
             // uncomment the following exception and declare AdminInterface as return type
             //
-            // throw new \LogicException(sprintf('Pool has no admin for the class %s.', $class));
+            // throw new AdminClassNotFoundException(sprintf('Pool has no admin for the class %s.', $class));
 
             return null;
         }
 
-        if (!$this->hasSingleAdminByClass($class)) {
-            throw new \RuntimeException(sprintf(
-                'Unable to find a valid admin for the class: %s, there are too many registered: %s',
+        if (isset($this->adminClasses[$class][self::DEFAULT_ADMIN_KEY])) {
+            return $this->getInstance($this->adminClasses[$class][self::DEFAULT_ADMIN_KEY]);
+        }
+
+        if (1 !== \count($this->adminClasses[$class])) {
+            throw new TooManyAdminClassException(sprintf(
```

Comment:
isn't this a BC break? Before it was not throwing anything and return the first one, or?

Reply:
But then the BC break would be that it was throwing `RuntimeException` before and now its throwing `TooManyAdminClassException` (which extends `InvalidArgumentException`)?